### ISSUE TITLE
Multiple email recipients in single notification broken

### ISSUE THREAD

**Issue by enkrs** [laravel/framework#24956]
- Laravel Version: 5.6.28

### Description:
Commit e41bee911df3bb07f7a903165626ead303fc76a9 introduces a way to specify recipient name for email. But the functionality breaks existing possible delivery to multiple users.

Existing functionality was that `routeNotificationFor('mail'` returns either a string with email `"aa@example.com"` or array `[0=>"aa@example.com", 1=>"bb@example.com"]`. The new functionality adds `["Aa"=>"aa@example.com","Bb"=>"bb@example.com"]` way to specify recipient names.

But because using `collect($recipients)->mapWithKeys`, the values without keys get overwritten: 
`[0=>"aa@example.com", 1=>"bb@example.com"]` becomes `[0=>"bb@example.com"]` (because of the way the mapped function works) and notification does not reach all recipients.

This is a bug, and I think some sort of workaround can still fix this.

maybe:
```
-            return [(is_string($recipient) ? $recipient : $recipient->email)];
+            return [$email => (is_string($recipient) ? $recipient : $recipient->email)];
```
(confusing variable names)

**Comment by enkrs** [comment]
Correction, The new functionality adds `["aa@example.com"=>"Aa", "bb@example.com"=>"Bb"]` way to specify recipient names. I will test out if my proposed solution passes tests.

**Comment by enkrs** [comment]
Fixed in 5.6.29
